src/promptflow/promptflow/batch/_base_executor_proxy.py

```diff
@@ -115,6 +117,17 @@ async def exec_aggregation_async(
         result = self._process_http_response(response)
         return AggregationResult.deserialize(result)
 
+    async def ensure_executor_startup(self, error_file):
+        """Ensure the executor service is initialized before calling the API to get the results"""
+        try:
+            await self.ensure_executor_health()
+        except ExecutorServiceUnhealthy as ex:
+            # raise the init error if there is any
+            init_ex = self._check_startup_error_from_file(error_file)
+            # TODO: will remove the destroy call after executor proxy creation is put into the run function
+            self.destroy()
```

Comment:
Add a log here to show ex? Also mention to destroy?

Reply:
Added.